sdk/samples/iot/vxworks/azureProvisioningClientSample.c

```diff
@@ -0,0 +1,996 @@
+}
+
+/*******************************************************************************
+ *
+ * deviceRegistrationStatusHandle - handle a device registration status response
+ *
+ * This routine handles a device registration status response.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void deviceRegistrationStatusHandle(
+    az_iot_provisioning_client* azureClient, /* provisioning client */
+    struct mosquitto* mqttClient, /* MQTT client object */
+    az_iot_provisioning_client_register_response* response /* response */
+)
+{
+  if (az_iot_provisioning_client_operation_complete(response->operation_status))
+  {
+    printf_s("Device registration is completed!\n");
+    if (response->operation_status == AZ_IOT_PROVISIONING_STATUS_ASSIGNED)
+    {
+      printf_s("Device is successfully provisioned!\n");
+    }
+    else
+    {
+      printf_s("Device provisioning is failed!\n");
+    }
+    (void)mosquitto_disconnect(mqttClient);
+  }
+  else
+  {
+    /*
+     * if the Registration is not finished, send a status query request to
+     * check again.
+     */
+    printf_s("Device registration is in progress!\n");
+    queryStatusSend(azureClient, mqttClient, response);
+  }
+}
+
+/*******************************************************************************
+ *
+ * onMessage - message receiving callback of a MQTT client
+ *
+ * This routine is called when a MQTT client receives a message.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void onMessage(
+    struct mosquitto* mqttClient, /* MQTT client object */
+    void* obj, /* user data provided */
+    /* in mosquitto_new */
+    const struct mosquitto_message* message /* message object */
+)
+{
+  az_span topic;
+  az_span msgSpan;
+  az_iot_provisioning_client* azureClient;
+  az_iot_provisioning_client_register_response registerRsp;
+
+  if (message != NULL)
+  {
+    topic = az_span_create_from_str(message->topic);
+    msgSpan = az_span_create((uint8_t*)(message->payload), message->payloadlen);
+  }
+  else
+  {
+    printf_s("onMessage: message is NULL!\n");
+    return;
+  }
+  if (obj != NULL)
+  {
+    azureClient = (az_iot_provisioning_client*)obj;
+  }
+  else
+  {
+    printf_s("onMessage: obj is NULL!\n");
+    return;
+  }
+
+  if (az_result_succeeded(az_iot_provisioning_client_parse_received_topic_and_payload(
+          azureClient, topic, msgSpan, &registerRsp)))
+  {
+    deviceRegistrationStatusHandle(azureClient, mqttClient, &registerRsp);
+  }
+}
+
+#ifndef X509_CERTIFICATE
+
+/*******************************************************************************
+ *
+ * base64Decode - decode a base64 string to a binary buffer
+ *
+ * This routine decodes a base64 string to a binary buffer.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: true if the decoding is successful, false otherwise
+ *
+ * \NOMANUAL
+ */
+
+static bool base64Decode(az_span base64Encoded, az_span* decoded)
```

Comment:
is there a way that this is not dupped between the two samples?

#include "utils.c"

Reply:
Actually, we have thought of putting base64Decode and base64Encode into a common file.
But we didn't do this because:

1. Currently, we only have 2 samples. The maintenance effort is not big. And we can have benefits in 2.
2. Less dependencies. The user can see all the code at a time instead of crossing several files. And they just need to copy one file to their project. It is very convenient for a freshman of Azure SDK to start his/her first try.
3. We see that on the latest "main" branch, there are "az_base64_encode" and "az_base64_decode". We are going to use these functions instead of implementing them again. But now VxWorks is aligned with release 1.1.0 which doesn't include these functions. So we are going to do the replacement when we do alignment to the new release.

Do you think it is OK?